### PR TITLE
PAINT-298: Update dependencies and fix ndk issues

### PR DESCRIPTION
**Pull request by ThomasSchwengler**
* Filter abi support to all supported abis by ndk `>= 17`
* Update dependencies
* ~Upgrade `buildToolsVersion`~
* Upgrade `compileSdkVersion` to 26
* Upgrade gradle wrapper
* Remove custom runner, this shouldn't be needed anymore with
  the new android emulator on jenkins
* Migrate dependency configuration to 3.0.0 ([ref](https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration))
* Fix `Jenkinsfile` apk location

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -21,18 +21,17 @@
...
     }
 }
 
 allprojects {
     repositories {
         jcenter()
-        maven {
-            url "https://maven.google.com"
-        }
+        maven { url 'https://maven.google.com' }
```
Comment (daberni) [inline build.gradle]: could be `google()` too

File: Paintroid/build.gradle [hunk 2]
```diff
@@ -41,18 +41,19 @@ if (project.hasProperty('independent')) {
 }
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.3'
+    compileSdkVersion 26
+    buildToolsVersion '26.0.2'
```
Comment (daberni) [inline Paintroid/build.gradle]: buildToolsVersion is ignored with android-gradle-plugin versions >= 3
this line can be removed

File: build.gradle [hunk 3]
```diff
@@ -21,18 +21,17 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.3'
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (redeamer) [inline build.gradle]: If the gradle version is upgraded, the output of the debug *.apk changes. There will be an additional subdirectory under `build/outputs/apk/` for the build flavor (eg debug). Please respect the new Location in the `Jenkinsfile`. It is defined inside the `APK_LOCATION_DEBUG` variable. It should change from `.../apk/Paintroid-debug.apk` to `.../apk/debug/Paintroid-debug.apk`

Comment (ThomasSchwengler) [reply build.gradle]: Should be fixed now

File: Paintroid/build.gradle (hunk 2 above)
Comment (ThomasSchwengler) [reply Paintroid/build.gradle]: Thanks, removed the `buildToolsVersion` line

File: build.gradle (hunk 1 above)
Comment (ThomasSchwengler) [reply build.gradle]: You are right, switched it to `google()`

**Pull request merged**